### DISCUSSION TITLE
Switching to fullscreen in macOS disables outer functionalities completely

### DISCUSSION THREAD

**Discussion by NafeeJ** [ppy/osu#13351] [Q&A] [answered]
When in fullscreen mode in settings (not the native macOS fullscreen green button thing or borderless), it seems I can't use the rest of the OS and have to exist osu to use or see anything else. Is there a way to get out of it or is it an issue thats not fixed yet?

Borderless isn't viable for me because it seems changing resolution doesn't work or is unimplemented yet and playing at native resolution makes my mac go to 80C with terrible FPS

I'm running version 2021.602.0 on a 2015 MacBook Pro i5 13"

**Comment by frenzibyte** [comment]
That can be done osu!framework-side by overriding the fullscreen switching in SDL2 window implementation and using native code to properly switch instead.

**Reply by peppy** [reply]
"native" is what borderless is already doing. you are misunderstanding this request, which is actually 99% impossible to achieve.

- supporting changing resolution is legacy functionality and doesn't play well with the window controls.
- supporting changing resolution is done to allow exclusive fullscreen. even if we could make the window controls display this would destroy all benefit of that.

**Reply by frenzibyte** [reply]
Right, then I believe this is not an issue then, because that's how macOS works and shouldn't be changed.

**Comment by frenzibyte** [comment, marked as answer]
As per https://github.com/ppy/osu/discussions/13351#discussioncomment-831131, this is not an issue that can be resolved on our side.
